### DISCUSSION TITLE
GPU Transcoding/ABR Benchmarks

### DISCUSSION THREAD

**Discussion by rsnk96** [ant-media/Ant-Media-Server#5310] [Q&A]
Hi AMS team and community

Has anyone done a benchmark of how many streams can be transcoded using AMS when using a given GPU?

Ex: AMS running on a single Nvidia Tesla T4 GPU (Ex: through a g4dn instance on AWS) may be able  to transcode `x` streams parallely down from 1080p-->240p

**Comment by lastpeony** [comment]
To the best of my knowledge, no, there isn't. However, it would be beneficial to have this benchmark. Perhaps @burak-58 can provide more insights and expertise on this particular matter.

**Comment by rsnk96** [comment]
We are an Enterprise customer, and this would help with some of our cost assessments.  Do let us know if we can help with the same!

**Reply by mekya** [reply]
Hi @rsnk96,
We have some results and it seems that we need to make them accessible for everyone. Furthermore, there was a pending PR that saves some CPU processing power by just giving the option to passthrough the audio.

I think yes your support will be very helpful and we can work together. 

Let's define the parameters may have an impact to the results. 

- What's the (bitrate) do you plan to use for publishing?
- Do you publish with RTMP or WebRTC?
- How many bitrates would be available?  
   - 1080p
   - 720p
   - 640p
   - 480p
   - 360p
   - 240p
 - Do you plan to use WebRTC for playback?

**Comment by rsnk96** [comment]
Hi @mekya 
Hope this helps, do lmk if I can help with any other info:
* **What's the (bitrate) do you plan to use for publishing?**: We usually publish at 1,2, and 4 Mbps. Apart from this, we also have low-bandwidth use-cases in which we publish at 150 kbps (which doesn't need any transcoding).
* **Do you publish with RTMP or WebRTC?**: Publish using RTMP
* **How many bitrates would be available? **: We would like all streams to have a 240/360p, 150kbps alternate sub-stream option available for low-bandwidth viewing
* **Do you plan to use WebRTC for playback?**: No, we do not. Reason: In our tests a couple of months back, we saw that WebRTC streams usually take a couple of seconds to begin, and hence, we rolled back to HLS streams.

**Reply by mekya** [reply]
hi @rsnk96,

Thank you for your comment. I just had a chance to respond.

- If you publish with 150Kbps, 1, 2 and 4mpbs different rates, it significantly changes which bitrate to choose. I mean if incoming stream is 360p, server does not scale to higher resolution because it's none sense. 

- So if we publish 360p and let have 240p/360p and 150k solutions and calculate the number of streams in one of g4dn instance, does that work for benchmark to give an idea to you? If not, please let me know which option you would like to use?

- Lastly how many simultaneous streams are you going to have in your system?

Cheers

**Reply by rsnk96** [reply]
Sorry, I missed one point: Input resolution can be both 360p and 720p.

Further, on your questions:

> So if we publish 360p and let have 240p/360p and 150k solutions and calculate the number of streams in one of g4dn instance, does that work for benchmark to give an idea to you? If not, please let me know which option you would like to use?

Yes, this works for publishing 1) 360p and 2) 720p


> 

> Lastly how many simultaneous streams are you going to have in your system?

50-200

**Reply by mekya** [reply]
Hi @rsnk96,

Thank you so much. It seems that we're working on this issue specificially for one case and I can prepare something for you as well. 
I've created this issue for that https://github.com/ant-media/Ant-Media-Server/issues/5433

After this benchmark, are you ready to use the Ant Media Server on your end?

**Reply by mekya** [reply]
I've added scenario 2 and 3 for you, please let me know if it's ok for you? @rsnk96 

https://github.com/ant-media/Ant-Media-Server/issues/5433

**Reply by rsnk96** [reply]
Perfect, they work @mekya , thanks. I'll watch the issue you created.
